optuna/_gp/gp.py

```diff
@@ -186,11 +166,54 @@ def loss_func(raw_params: np.ndarray) -> tuple[float, np.ndarray]:
         loss.backward()  # type: ignore
         return loss.item(), raw_params_tensor.grad.detach().numpy()  # type: ignore
 
+    default_initial_kernel_params = KernelParamsTensor(
+        inverse_squared_lengthscales=torch.ones(n_params, dtype=torch.float64),
+        kernel_scale=torch.tensor(1.0, dtype=torch.float64),
+        noise_var=torch.tensor(1.0, dtype=torch.float64),
+    )
+    if initial_kernel_params is None:
+        initial_kernel_params = default_initial_kernel_params
+
     # jac=True means loss_func returns the gradient for gradient descent.
-    res = so.minimize(loss_func, initial_raw_params, jac=True)
+    def optimize(initial_kernel_params: KernelParamsTensor) -> np.ndarray:
```

Comment:
By renaming `fit_kernel_params` to `_fit_kernel_params` and introducing the following thin wrapper, any future modifications in `_fit_kernel_params` will be easier:

```python
def fit_kernel_params(...) -> KernelParamsTensor:
    default_initial_kernel_params = KernelParamsTensor(...)
    if initial_kernel_params is None:
        initial_kernel_params = default_initial_kernel_params

    for init_kernel_params in [initial_kernel_params, default_initial_kernel_params]:
        try:
            return _fit_kernel_params(..., init_kernel_params)
        except RuntimeError:
            pass

    return default_initial_kernel_params
```

In this way, it is also easier to modify the part, which makes the optimization robust, in the future as well.

Please take a look at [here](https://github.com/nabenabe0928/optuna/compare/master...nabenabe0928:optuna:code-fix/refactor-fit-kernel-params?expand=1) for more details.